File: docker_sdk/python/docker_sdk/model/project.py
from typing import List
from sqlalchemy import String, Integer, Boolean, DateTime, func, JSON
from sqlalchemy.orm import Mapped, mapped_column, relationship
from docker_sdk.model.base import Base


class Project(Base):
    __tablename__ = 'projects'

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(30), unique=True, nullable=False)
    path: Mapped[str] = mapped_column(String(30), nullable=False)
    latest_bootable: Mapped[Boolean] = mapped_column(Boolean, default=False)
    status: Mapped[int] = mapped_column(Integer, default=0)
    data: Mapped[JSON] = mapped_column(JSON, nullable=True)
    created_at: Mapped[DateTime] = mapped_column(DateTime, nullable=False, default=func.now())
    updated_at: Mapped[DateTime] = mapped_column(DateTime, nullable=False, default=func.now(), onupdate=func.now())

    shared_services: Mapped[List["SharedServiceProject"]] = relationship(
        back_populates='project'
    )

    service_endpoints: Mapped[List["ProjectServiceEndpoint"]] = relationship(
        back_populates='project'
    )

    storage: Mapped["ProjectStorage"] = relationship(
        back_populates='project',
        uselist=False
    )

File: docker_sdk/php/docker-sdk/src/Model/Project.php
<?php

/**
 * This file is part of the Spryker Suite.
 * For full license information, please view the LICENSE file that was distributed with this source code.
 */

namespace DockerSDK\Model;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Project extends Model
{
    protected $table = 'projects';

    protected $fillable = [
        'name',
        'path',
        'latest_bootable',
        'status',
        'data',
        'created_at',
        'updated_at',
    ];

    protected $casts = [
        'latest_bootable' => 'boolean',
    ];

    public function sharedServices(): BelongsToMany
    {
        return $this->belongsToMany(
            SharedService::class,
            'shared_services_projects',
            'project_id',
            'shared_service_id'
        )->using(
            SharedServiceProject::class
        )->withPivot(['data']);
    }

    public function projectServiceEndpoints(): HasMany
    {
        return $this->hasMany(ProjectServiceEndpoint::class);
    }

    public function storage(): HasOne
    {
        return $this->hasOne(ProjectStorage::class);
    }
}
